code/modules/tooltip/tooltip.dm

```diff
@@ -0,0 +1,130 @@
+/*
+Tooltips v1.1 - 22/10/15
+Developed by Wire (#goonstation on irc.synirc.net)
+- Added support for screen_loc pixel offsets. Should work. Maybe.
+- Added init function to more efficiently send base vars
+
+Configuration:
+- Set control to the correct skin element (remember to actually place the skin element)
+- Set file to the correct path for the .html file (remember to actually place the html file)
+- Attach the datum to the user client on login, e.g.
+	/client/New()
+		src.tooltips = new /datum/tooltip(src)
+
+Usage:
+- Define mouse event procs on your (probably HUD) object and simply call the show and hide procs respectively:
+	/obj/screen/hud
+		MouseEntered(location, control, params)
+			usr.client.tooltip.show(params, title = src.name, content = src.desc)
+
+		MouseExited()
+			usr.client.tooltip.hide()
+
+Customization:
+- Theming can be done by passing the theme var to show() and using css in the html file to change the look
+- For your convenience some pre-made themes are included
+
+Notes:
+- You may have noticed 90% of the work is done via javascript on the client. Gotta save those cycles man.
+- This is entirely untested in any other codebase besides goonstation so I have no idea if it will port nicely. Good luck!
+	- After testing and discussion (Wire, Remie, MrPerson, AnturK) ToolTips are ok and work for /tg/station13
+*/
+
+
+/datum/tooltip
+	var/client/owner
+	var/control = "mainwindow.tooltip"
+	var/showing = FALSE
+	var/queueHide = FALSE
+	var/init = FALSE
+
+
+/datum/tooltip/New(client/C)
+	if (!C)
+		stack_trace("/datum/tooltip/New() called without a valid client")
+		return ..()
+	owner = C
+	var/datum/asset/stuff = get_asset_datum(/datum/asset/simple/jquery)
+	stuff.send(owner)
+	
+	owner << browse(file2text('code/modules/tooltip/tooltip.html'), "window=[control]")
+
+	return ..()
+
+
+/datum/tooltip/proc/show(atom/movable/thing, params = null, title = null, content = null, theme = "default", special = "none")
+	if (!thing || !params || (!title && !content) || !owner || !isnum(world.icon_size))
+		return FALSE
+	if (!init)
+		//Initialize some vars
+		init = TRUE
+		owner << output(list2params(list(world.icon_size, control)), "[control]:tooltip.init")
+
+	showing = TRUE
+
+	if (title && content)
```

Comment:
```
if(title)
	if(content)
		title = "<h1>[title]</h1>"
		content = "<p>[content]</p>"
	else
		title = "<p>[title]</p>"
else if(content)
	content = "<p>[content]</p>"
```

Reply:
I'd rather just force a title as h1 and content as p and force the contributor to code it correctly, then we can just do
```
if(title)
    title = "<h1>[title]</h1>"
if(content)
    content = "<p>[content]</p>"
```

Reply:
Up to you.